caikit/core/model_management/local_model_initializer.py

```diff
@@ -177,8 +177,10 @@ def init(self, model_config: ModuleConfig, **kwargs) -> Optional[ModuleBase]:
                     module_backend_impl.__name__,
                 )
                 extra_kwargs = {}
-                if self._supports_load_backend_kwarg(module_backend_impl.load):
+                if self._supports_arg(module_backend_impl.load, "load_backend"):
                     extra_kwargs["load_backend"] = load_backend
+                if self._supports_arg(module_backend_impl.load, "model_config"):
```

Comment:
question: The idea would be that the underlying module can accept `model_config` as an argument to `load` function to find the model mapping in remote cases?

Reply:
I think this would also mean that the caikit-module should not be having any sort of validation or configuration options which rely on reading the actual model's config file right?

Reply:
Wondering how does this correlate to using HF `config.json` as finder?

Reply:
Yeah, exactly. When we did the `finder`/`initializer` split, the interface actually got a bit awkward. Ideally, a `finder` returns a `ModuleConfig` and an `initializer` takes that `ModuleConfig` and is able to load directly. If it _requires_ on-disk assets (e.g. weights), there's no way around it, but ideally the `config.yml` would not need to be read from disk. The problem is that every `def load` function relies on being passed a `model_path` which is assumed to be the parent directory of the `config.yml`. Having `model_config` as an arg allows modules that are able to load without a `config.yml` to do so whereas now, the only way to get back that coordination information is to read `config.yml` or "just know it" (i.e. look for a `config.json` and parse it).

Reply:
Wondering if an alternate implementation of converting the `model_config` here loadable / readable automatically by `ModuleConfig` itself make it opaque to modules and require no change in like TGIS module etc?